tests/config.go

```diff
@@ -0,0 +1,91 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2019 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+var ConfigFile = ""
+var Config *KubeVirtTestsConfiguration
+
+func init() {
+	flag.StringVar(&ConfigFile, "config", "tests/default-config.json", "Path to a JSON formatted file from which the test suite will load its configuration. The path may be absolute or relative; relative paths start at the current working directory.")
+}
+
+// KubeVirtTestsConfiguration contains the configuration for KubeVirt tests
+type KubeVirtTestsConfiguration struct {
+	// StorageClass to use to create local PVCs
+	StorageClassLocal string `json:"storageClassLocal"`
+	// StorageClass to use to create host-path PVCs
+	StorageClassHostPath string `json:"storageClassHostPath"`
+	// StorageClass to use to create block-volume PVCs
+	StorageClassBlockVolume string `json:"storageClassBlockVolume"`
+	// StorageClass to use to create rhel PVCs
+	StorageClassRhel string `json:"storageClassRhel"`
+	// StorageClass to use to create windows PVCs
+	StorageClassWindows string `json:"storageClassWindows"`
+}
+
+func NewKubeVirtTestsConfiguration() *KubeVirtTestsConfiguration {
+	return &KubeVirtTestsConfiguration{
```

Comment:
I think we can remove this default config now, because we always read from file and panic when it goes wrong

Reply:
I hesitated and decided to keep it in case the user provides a partial configuration file overriding just a few values.

Reply:
Ah ok, I assumed you have to always provide all values. But overriding a few values only makes sense.
However, now we have the default values at 2 places, in code and in file, which is confusing :/
What about this: remove the default config from code, and remove the default value of the config flag, but still always load `tests/default-config.json` first. And additionally load a custom config file afterwards when the config flag is given?

Reply:
Sounds smart, I pushed a new commit to implement this.